hypershift-operator/controllers/nodepool/azure.go

```diff
@@ -49,6 +52,25 @@ func azureMachineTemplateSpec(hcluster *hyperv1.HostedCluster, nodePool *hyperv1
 		FailureDomain: failureDomain(nodePool),
 	}}}
 
+	switch nodePool.Spec.Platform.Azure.Image.Type {
```

Comment:
can all this be a unit tested function?

Reply:
Yeah I added a unit test for the whole function with several scenarios.